diff --git a/src/lib.rs b/src/lib.rs
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -399,6 +399,8 @@ pub async fn run<D>(device: D, mtu: u16, args: Args, shutdown_token: Cancellatio
                     }
                     Err(e) => {
                         log::error!("Failed to create UDP connection: {e}");
+                        let count = task_count.fetch_sub(1, Relaxed).saturating_sub(1);
+                        log::trace!("Session count {count}");
                     }
                 }
             }
